library_management/src/main/java/com/group2/library_management/service/impl/BookInstanceServiceImpl.java

```diff
@@ -25,4 +39,20 @@ public List<BookInstanceResponse> getBookInstancesByEditionId(Integer editionId)
                .map(bookInstanceMapper::toDto)
                .collect(Collectors.toList());
     }
+
+    @Override
+    public void deleteBookInstanceByBookInstanceId(Integer bookInstanceId) throws BookInstanceNotFoundException, CannotDeleteResourceException{
+        BookInstance bookInstance = bookInstanceRepository.findById(bookInstanceId).orElseThrow(() -> new BookInstanceNotFoundException(bookInstanceId));
+        if(checkStatusBookInstance(bookInstance.getStatus().toString())){
+            throw new CannotDeleteResourceException(getMessage(bookInstance.getBarcode(), enumDisplayService.getBookStatusDisplayName(bookInstance.getStatus())));
+        }
+        bookInstanceRepository.delete(bookInstance);
+    }
+
+    private Boolean checkStatusBookInstance(String status){
+        if (status.equals("AVAILABLE") || status.equals("BORROWED") || status.equals("RESERVED") || status.equals("REPAIRING")) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
```suggestion
    private static final Set<String> NON_DELETABLE_STATUSES = Set.of(
    "AVAILABLE", "BORROWED", "RESERVED", "REPAIRING"
);

private Boolean checkStatusBookInstance(String status) {
    return NON_DELETABLE_STATUSES.contains(status);
}
```

Nên khai báo beiens hằng nhé